### PR TITLE
feat: add EventStorage & EventInteractor

### PR DESCRIPTION
**Pull request by yshrsmz**

### CODE REVIEW

File: src/internal/event/EventInteractor.ts [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+      event: newDefaultEvaluationEvent(
+        newBaseEvent(this.clock.currentTimeSeconds(), newMetadata(this.appVersion, '', '')),
+        {
+          featureId,
+          featureVersion: 0,
+          variationId: '',
+          userId: user.id,
+          user,
+          reason: {
+            type: 'DEFAULT'
```
Comment (cre8ivejp) [inline src/internal/event/EventInteractor.ts:63]: Please use `CLIENT` instead. The `DEFAULT` is used on the server side.

File: src/internal/model/Event.ts [hunk 2]
```diff
@@ -11,37 +11,33 @@ export const EventType = {
 } as const
 export type EventType = typeof EventType[keyof typeof EventType]
 
-export interface GoalEvent {
+export interface BaseEvent {
```
Comment (cre8ivejp) [inline src/internal/model/Event.ts:14]: 👍

File: src/internal/event/EventInteractor.ts (hunk 1 above)
Comment (yshrsmz) [reply src/internal/event/EventInteractor.ts]: fixed in [`7346096` (#26)](https://github.com/bucketeer-io/javascript-client-sdk/pull/26/commits/734609694f9c886db4656d14b46a5ec4c43f8cec)

**Review by cre8ivejp — APPROVED** [review]
Thank you!

**Pull request merged**